src/App.vue

```diff
@@ -1,29 +1,54 @@
 <template>
-  <div id="app">
-    <img
-      alt="Vue logo"
-      src="./assets/logo.png"
-    />
-    <HelloWorld msg="Hello Vue 2 + Vite" />
+  <div>
+    <the-header></the-header>
+    <main>
+      <product-list :product-list="productList"></product-list>
+    </main>
+    <the-footer></the-footer>
   </div>
 </template>
 
 <script>
-import HelloWorld from "./components/HelloWorld.vue";
+import TheHeader from "./components/layout/TheHeader.vue";
+import TheFooter from "./components/layout/TheFooter.vue";
+import ProductList from "./components/product/ProductList.vue";
+
 export default {
   components: {
-    HelloWorld,
+    TheHeader,
+    TheFooter,
+    ProductList,
+  },
+  data() {
+    return {
+      productList: [],
+      justText: "Text here",
+    };
+  },
+  provide() {
+    return {
+      fetchProductData: this.fetchProductData,
+    };
+  },
+  methods: {
+    async fetchProductData() {
+      const response = await fetch("api/products");
+
+      if (!response.ok) {
```

Comment:
Do you have any top level handler for exceptions?  
if not, I'd say that in this case you could both handle an api error with for example a `try catch` block and maybe log the error into the console or, even better, show a message that that something went wrong to the user. 😃